File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> ans = new ArrayList<>();
        HashMap<String, List<String>> map = new HashMap<>();
        for(String st:strs){
            char[] s1 = st.toCharArray();
            Arrays.sort(s1);
            StringBuilder sb = new StringBuilder();
            for(int i=0;i<s1.length;i++){
                sb.append(s1[i]);
            }
            String s2 = sb.toString();
            if(!map.containsKey(s2)){
                map.put(s2, new ArrayList<>());
            }
            map.get(s2).add(st);
        }
        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        adict = {}

        for word in strs:
            sorted_word = ''.join(sorted(word))
            if sorted_word not in adict:
                adict[sorted_word] = []
            adict[sorted_word].append(word)

        return list(adict.values())
        